src/ddscxx/tests/SharedMemory.cpp

```diff
@@ -60,7 +65,42 @@ void make_sample_(Bounded::Msg & sample, const int32_t cnt)
   sample.bounded_sequence().reserve(255);
   std::fill(sample.bounded_sequence().begin(), sample.bounded_sequence().begin() + 255, cnt);
 }
+
+template<typename T>
+struct MessageProperty;
+
+template<>
+struct MessageProperty<Space::Type1> 
+{
+  static constexpr char INSTANCE_NAME[] = "Space::Type1";
+  static constexpr bool IS_SHARED_MEMORY_COMPATIBLE = true;
```

Comment:
Similarly we can get this info from the type using such a trait `org::eclipse::cyclonedds::topic::TopicTraits<DDSType>::isSelfContained()`

If we use these traits, we then probably will not need this `MessageProperty` struct